scripts/ci/sync_extensions.py

```diff
@@ -62,20 +62,27 @@ def _sync_wheel(ext, updated_indexes, failed_urls, overwrite, temp_dir):
     if not overwrite:
         cmd = ['az', 'storage', 'blob', 'exists', '--container-name', f'{STORAGE_CONTAINER}', '--account-name',
                f'{STORAGE_ACCOUNT}', '--name', f'{blob_name}', '--auth-mode', 'login']
-        result = subprocess.run(cmd, check=True, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
Per https://docs.python.org/3/library/subprocess.html#subprocess.run, `stdout=subprocess.PIPE, stderr=subprocess.STDOUT` combines both streams into one:

> If you wish to capture and combine both streams into one, use `stdout=PIPE` and `stderr=STDOUT` instead of `capture_output`.

Reply:
If I use result = subprocess.run(cmd, check=True, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT),
The output of result.stdout is like this:: 
![image](https://github.com/Azure/azure-cli-extensions/assets/18628534/eda950d9-8f5e-4371-87ad-c3a603c4d8b0)
https://dev.azure.com/azclitools/internal/_build/results?buildId=128429&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=3178dee2-26ab-587a-4407-0f5d5c54e1f3&l=82

Reply:
I can reproduce on Linux:

```py
import subprocess
result = subprocess.run(['az', 'account', 'show'], check=True, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
print(result)
```

`shell=True` is causing problem.

https://docs.python.org/3/library/subprocess.html#popen-constructor

> On POSIX with shell=True, the shell defaults to /bin/sh. ... If args is a sequence, the first item specifies the command string, and any additional items will be treated as additional arguments to the shell itself.

So `account`, `show` are actually passed to the shell, not `az`.

Reply:
`shell=True` comes from #7263 and breaks the `az` command on Linux. That's why in https://github.com/Azure/azure-cli-extensions/pull/7263#issuecomment-1931550231 I mentioned it is better to fully test the script before merging it to `main`.